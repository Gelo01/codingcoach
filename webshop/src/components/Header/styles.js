import styled from 'styled-components';

export const InnerContainer = styled.div`
  align-items: center;
  display: flex;
`;
