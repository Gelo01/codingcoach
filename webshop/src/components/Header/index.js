import React from 'react';

import { InnerContainer } from './styles';

const Header = () => {
  return (
    <header>
      <InnerContainer>
        Header
      </InnerContainer>
    </header>
  );
}

export default Header;
