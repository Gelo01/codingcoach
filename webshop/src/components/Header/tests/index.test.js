import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';

import Header from '..';

describe('Header', () => {
  it('Should render correctly', () => {
    const { container } = render(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should update the title accordingly', () => {
    const { getByTestId } = render(<Header />);

    const input = getByTestId('updateTitle-input');

    fireEvent.change(input, { currentTarget: { value: 'some title' } });

    // This is not a very common test and this setTimeout trick is kinda hacky
    // It's just because the state change is asynchronous, so I'm doing this
    // to wait for state changes before proceeding with the test
    setTimeout(() => {
      const headerTitle = getByTestId('title');

      expect(headerTitle.textContent).toEqual('some title');
    }, 1000);

    
  });

  it('Should not show the container onMount', () => {
    const { queryByTestId } = render(<Header />);

    expect(queryByTestId('container')).toBeFalsy();
  });

  it('Should show the container if the title changes to "interstellar"', async () => {
    const { getByTestId, queryByTestId } = render(<Header />);

    const input = getByTestId('updateTitle-input');

    fireEvent.change(input, { currentTarget: { value: 'interstellar' } });

    const containerIsBeingShown = waitForElement(() => queryByTestId('container'));

    expect(containerIsBeingShown).toBeTruthy();
  });
});
