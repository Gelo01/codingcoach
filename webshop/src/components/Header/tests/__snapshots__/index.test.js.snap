// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<header>
  <div
    class="c0"
  >
    Header
  </div>
</header>
`;
